$NetBSD$
- protect Xen's stdarg.h from multiple inclusion.
- build fix for stdarg under NetBSD.

Both backported from upstream.

http://xenbits.xensource.com/hg/xen-unstable.hg/file/a574bf2f5059/xen/include/xen/stdarg.h
--- xen/include/xen/stdarg.h.orig	2009-08-06 12:56:43.000000000 +0000
+++ xen/include/xen/stdarg.h
@@ -1,5 +1,15 @@
+#ifndef __XEN_STDARG_H__
+#define __XEN_STDARG_H__
+
 #if defined(__OpenBSD__)
 #  include "/usr/include/stdarg.h"
+#elif defined (__NetBSD__)
+   typedef __builtin_va_list va_list;
+#  define va_start(ap, last)    __builtin_stdarg_start((ap), (last))
+#  define va_end(ap)            __builtin_va_end(ap)
+#  define va_arg                __builtin_va_arg
 #else
 #  include <stdarg.h>
 #endif
+
+#endif /* __XEN_STDARG_H__ */
